app/models/concerns/indexable.rb

```diff
@@ -18,5 +20,19 @@ module Indexable
     after_commit on: [:destroy] do
       RemoveFromIndexJob.perform_async(self.class.name, id)
     end
+
+    def set_published_at
+      unless published_at
+        if defined?(metadata) && metadata && metadata['date']
+          begin
+            update_column(:published_at, metadata['date'])
+          rescue ActiveRecord::StatementInvalid
```

Comment:
Hmm. We may need to address that in a future PR. I think we should be able to handle a year date for books, as many of them will only have a year, I believe. Can you create a separate issue and add it to GH?